mirgecom/multiphysics/thermally_coupled_fluid_wall.py

```diff
@@ -496,18 +752,32 @@ def get_interface_boundaries(
                 fluid_state.array_context, dcoll, fluid_dd)
             * (0*fluid_state.temperature+1))
 
-        fluid_interface_boundaries = {
-            kappa_tpair.dd.domain_tag: InterfaceFluidBoundary(
-                kappa_tpair.ext,
-                temperature_tpair.ext,
-                grad_temperature_tpair.ext,
-                wall_penalty_amount,
-                lengthscales=op.project(dcoll,
-                    fluid_dd, temperature_tpair.dd, fluid_lengthscales))
-            for kappa_tpair, temperature_tpair, grad_temperature_tpair in zip(
-                kappa_inter_vol_tpairs[wall_dd, fluid_dd],
-                temperature_inter_vol_tpairs[wall_dd, fluid_dd],
-                grad_temperature_inter_vol_tpairs[wall_dd, fluid_dd])}
+        if interface_noslip:
+            fluid_interface_boundaries = {
+                kappa_tpair.dd.domain_tag: InterfaceFluidBoundary(
+                    kappa_tpair.ext,
+                    temperature_tpair.ext,
+                    grad_temperature_tpair.ext,
+                    wall_penalty_amount,
+                    lengthscales=op.project(dcoll,
+                        fluid_dd, temperature_tpair.dd, fluid_lengthscales))
+                for kappa_tpair, temperature_tpair, grad_temperature_tpair in zip(
+                    kappa_inter_vol_tpairs[wall_dd, fluid_dd],
+                    temperature_inter_vol_tpairs[wall_dd, fluid_dd],
+                    grad_temperature_inter_vol_tpairs[wall_dd, fluid_dd])}
+        else:
+            fluid_interface_boundaries = {
+                kappa_tpair.dd.domain_tag: InterfaceFluidSlipBoundary(
+                    kappa_tpair.ext,
+                    temperature_tpair.ext,
+                    grad_temperature_tpair.ext,
+                    wall_penalty_amount,
+                    lengthscales=op.project(dcoll,
+                        fluid_dd, temperature_tpair.dd, fluid_lengthscales))
+                for kappa_tpair, temperature_tpair, grad_temperature_tpair in zip(
+                    kappa_inter_vol_tpairs[wall_dd, fluid_dd],
+                    temperature_inter_vol_tpairs[wall_dd, fluid_dd],
+                    grad_temperature_inter_vol_tpairs[wall_dd, fluid_dd])}
```

Comment:
I think this can be simplified to something like:

```suggestion
        if interface_noslip:
            fluid_bc_class = InterfaceFluidBoundary
        else:
            fluid_bc_class = InterfaceFluidSlipBoundary

        fluid_interface_boundaries = {
            kappa_tpair.dd.domain_tag: fluid_bc_class(
                kappa_tpair.ext,
                temperature_tpair.ext,
                grad_temperature_tpair.ext,
                wall_penalty_amount,
                lengthscales=op.project(dcoll,
                    fluid_dd, temperature_tpair.dd, fluid_lengthscales))
            for kappa_tpair, temperature_tpair, grad_temperature_tpair in zip(
                kappa_inter_vol_tpairs[wall_dd, fluid_dd],
                temperature_inter_vol_tpairs[wall_dd, fluid_dd],
                grad_temperature_inter_vol_tpairs[wall_dd, fluid_dd])}
```